subprojects/files/src/main/java/org/gradle/api/internal/file/collections/DefaultFileCollectionResolveContext.java

```diff
@@ -121,16 +119,15 @@ public final ResolvableFileCollectionResolveContext newContext() {
             } else if (element instanceof TaskOutputs) {
                 TaskOutputs outputs = (TaskOutputs) element;
                 queue.add(0, outputs.getFiles());
-            } else if (element instanceof Callable) {
-                Callable callable = (Callable) element;
-                Object callableResult = uncheckedCall(callable);
-                if (callableResult != null) {
-                    queue.add(0, callableResult);
-                }
             } else if (element instanceof Provider) {
```

Comment:
:x: this branch is handled by `DeferredUtil` below and should be removed

Reply:
Right. The comparison to `null` made me postpone this but upon closer inspection it makes sense. Thanks.